airflow/api_connexion/endpoints/task_instance_endpoint.py

```diff
@@ -14,46 +14,260 @@
+    TaskInstanceCollection,
+    task_instance_collection_schema,
+    task_instance_schema,
+    task_instance_batch_form,
+    task_instance_reference_collection_schema,
+    TaskInstanceReferenceCollection,
+)
+from airflow.configuration import conf
+from airflow.models.dagrun import DagRun as DR
+from airflow.models.taskinstance import clear_task_instances, TaskInstance as TI
+from airflow.models.dag import DagModel
+from airflow.models.dagbag import DagBag
+from airflow.models import SlaMiss
+from airflow.utils.state import State
+from airflow.utils.session import provide_session
 
 
-@security.requires_access([("can_read", "Dag"), ("can_read", "DagRun"), ("can_read", "Task")])
-def get_task_instance():
-    """
-    Get a task instance
-    """
-    raise NotImplementedError("Not implemented yet.")
+@security.requires_access(
+    [
+        ("can_read", "Dag"),
+        ("can_read", "DagRun"),
+        ('can_read', 'Task'),
+    ]
+)
+@provide_session
+def get_task_instance(dag_id: str, dag_run_id: str, task_id: str, session=None):
+    """Get task instance"""
+    query = (
+        session.query(TI)
+        .filter(TI.dag_id == dag_id)
+        .join(DR, and_(TI.dag_id == DR.dag_id, TI.execution_date == DR.execution_date))
+        .filter(DR.run_id == dag_run_id)
+        .filter(TI.task_id == task_id)
+        .join(
+            SlaMiss,
+            and_(
+                SlaMiss.dag_id == TI.dag_id,
+                SlaMiss.execution_date == TI.execution_date,
+                SlaMiss.task_id == TI.task_id,
+            ),
+            isouter=True,
+        )
+        .add_entity(SlaMiss)
+    )
+    task_instance = query.one_or_none()
+    if task_instance is None:
+        raise NotFound("Task instance not found")
 
+    return task_instance_schema.dump(task_instance)
 
-@security.requires_access([("can_read", "Dag"), ("can_read", "DagRun"), ("can_read", "Task")])
-def get_task_instances():
-    """
-    Get list of task instances of DAG.
-    """
-    raise NotImplementedError("Not implemented yet.")
+
+def _apply_array_filter(query, key, values):
+    if values is not None:
+        query = query.filter(key.in_(values))
+    return query
+
+
+def _apply_range_filter(query, key, value_range: Tuple[Any, Any]):
+    gte_value, lte_value = value_range
+    if gte_value is not None:
+        query = query.filter(key >= gte_value)
+    if lte_value is not None:
+        query = query.filter(key <= lte_value)
+    return query
 
 
+@format_parameters(
+    {
+        "execution_date_gte": format_datetime,
+        "execution_date_lte": format_datetime,
+        "start_date_gte": format_datetime,
+        "start_date_lte": format_datetime,
+        "end_date_gte": format_datetime,
+        "end_date_lte": format_datetime,
+    }
+)
 @security.requires_access(
     [
-        ('can_read', 'Dag'),
-        ('can_read', 'DagRun'),
+        ("can_read", "Dag"),
+        ("can_read", "DagRun"),
         ('can_read', 'Task'),
```

Comment:
Can we change single to double quotes here too for consistency

Reply:
Fixed. :-)